tasks/winbuildscripts/Generate-Pyzip.ps1

```diff
@@ -61,45 +87,61 @@ if($maj -eq "2") {
 }
 elseif ($maj -eq "3") {
     $Env:PATH="$Env:PATH;c:\tools\msys64\mingw64\bin"
-    $p = Start-Process $dlfullpath -ArgumentList "/quiet DefaultJustForMeTargetDir=c:\pythonroot TARGETDIR=c:\pythonroot AssociateFiles=0 Include_doc=0 Include_launcher=0 Include_pip=0 Include_tcltk=0" -Wait -Passthru
 
+    try {
+        $installedPythonVersion = python --version
+        $installedPythonVersion = [regex]::match($installedPythonVersion,'Python (\d.\d.\d)').Groups[1].Value
+        $installedMaj, $installedMin, $installedPatch = $installedPythonVersion.Split(".")
+    
+        Write-Host -ForegroundColor Green "Detected installed Python $installedMaj $installedMin $installedPatch"
+
+        if ($installedMaj -ne "3") {
+            Write-Host -ForegroundColor Red "Invalid Python $installedPythonVersion installation detected"
+            exit 1   
+        }
+
+        # Python is already installed in the buildimage, but is the wrong version uninstall it
+        Get-Installer -url (Get-Installer-Url -version $installedPythonVersion) -dlfullpath $dlfullpath -md5sum $null
+        Write-Host -ForegroundColor Yellow "Uninstalling Python $installedPythonVersion"
+        $p = Start-Process $dlfullpath -ArgumentList "/quiet /uninstall" -Wait -Passthru
+        if ($p.ExitCode -ne 0) {
+            Write-Host -ForegroundColor Red ("Failed to uninstall Python, exit code 0x{0:X}" -f $p.ExitCode)
+            exit 1
+        }
+        Remove-Item $dlfullpath
+    }
+    catch {
+        # Python command not found
+    }
+
+    Get-Installer -url $url -dlfullpath $dlfullpath -md5sum $md5sum
+    Write-Host -ForegroundColor Green "Installing Python $Version"
+    $p = Start-Process $dlfullpath -ArgumentList "/quiet /log C:\mnt\install.log DefaultJustForMeTargetDir=""$pythonRoot"" DefaultAllUsersTargetDir=""$pythonRoot"" AssociateFiles=0 Include_doc=0 Include_launcher=0 Include_pip=0 Include_tcltk=0" -Wait -Passthru
```

Comment:
Out of curiosity: the previous code set the `TARGETDIR` variable, but here `DefaultAllUsersTargetDir` is set instead. What's the difference?